front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts

```diff
@@ -139,18 +200,32 @@ const applyLayout = (state: MacroEditorState) => {
   const scaleY = 500;
   const padding = 0.1;
 
+  // Default positions: based on long/lat.
+  const nodes: [NodeIndexed, number, number][] = [];
   for (const n of indexedNodes) {
     if (!n.dbId && n.geocoord !== undefined) {
       const normalizedX = (n.geocoord.lng - minX) / (width || 1);
       const normalizedY = 1 - (n.geocoord.lat - minY) / (height || 1);
       const paddedX = normalizedX * (1 - 2 * padding) + padding;
       const paddedY = normalizedY * (1 - 2 * padding) + padding;
-      state.updateNodeDataByKey(n.path_item_key, {
-        position_x: Math.round(scaleX * paddedX),
-        position_y: Math.round(scaleY * paddedY),
-      });
+      const positionX = Math.round(scaleX * paddedX);
+      const positionY = Math.round(scaleY * paddedY);
+      nodes.push([n, positionX, positionY]);
```

Comment:
This only takes into account nodes that we have a geo position for. Other nodes' position isn't updated to not avoid overlaps. I think we should adjust these other nodes as well.

Some nodes might be missing a geo position, e.g. when a node doesn't match an OP in the infra. This is in fact pretty common for macro studies.

Reply:
That's not exactly obvious, here: https://github.com/OpenRailAssociation/osrd/blob/4d95b3c29658f4655c3e86565cf9417eaa7965b6/front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts#L213